### PR TITLE
Refactor generated files

### PR DESCRIPTION
**Pull request by tvandijck**
depends on: https://github.com/premake/premake-core/pull/965

### CODE REVIEW

**Comment by tvandijck** [comment]
all this, to get this to work properly:
```lua
project 'tag_edit_codegen'
    kind 'Utility'

    -- remove all configs and just map everything to a single codegen config.
    removeconfigurations { '*' }
    configurations { 'Codegen' }
    configmap {
        ["*"] = "Codegen",
    }

    -- add all tag files.
    files { '../include/**.tag' }

    -- setup compile steps for all tag files.
    bindirdependencies { 'tagc' }
    filter { 'files:../include/**.tag' }
        buildmessage 'Compiling %{file.abspath}'
        buildcommands {
            'tagc --cpp --types --schema --out=%{sln.location}/generated_tag/tag_edit %{file.abspath}'
        }
        buildoutputs {
            '%{sln.location}/generated_tag/tag_edit/%{file.basename}.h',
            '%{sln.location}/generated_tag/tag_edit/%{file.basename}.cpp',
        }
        compilebuildoutputs 'true'
        compilebuildoutputstarget 'tag_edit'


project 'tag_edit'
    kind 'StaticLib'

    dependson 'tag_edit_codegen'
    includedependencies { 'tag', 'blz', 'bc' }
    includedirs {
        '../include',
        '%{sln.location}/generated_tag'
    }

    files { '*.h', '*.cpp', '../include/**' }

    vpaths {
        ['include/*'] = { '../include/**' },
        ['src/*']     = { '**' }
    }
```

**Comment by tvandijck** [comment]
I think this somehow also depended on #969, I just don't see how...
it seems to build and run the tests successfully here at blizzard, but fails here.... 
I will look at it if it fails again..

**Comment by tvandijck** [comment]
As this PR is based out of the Blizzard org branches, I have to pull these over to my own fork, close this PR and make a new one. Here at DICE we're not using premake, so this will have to become a hobby kind of thing. My computer at home is in a container somewhere on the ocean right now, so I won't have access to that until mid february... so please have a little patience.

**Comment by starkos** [comment]
#969 is merged now…is this still a valid PR, or should it be closed as mentioned in the comments?

**Comment by tvandijck** [comment]
No please leave open, I'll get to fixing this somewhere maybe this week/weekend.

**Comment by starkos** [comment]
Sounds good…didn't mean to rush, just trying to tidy up a bit.

**Comment by tvandijck** [comment]
This works now, and actually fixes a long standing issue where you could never reliably use `cfg.objdir` in file filters.

File: tests/oven/test_compilebuildoutputstarget.lua [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	local function prepare(buildcfg, platform)
+		cfg = test.getconfig(prj, buildcfg, platform)
+	end
+
+	function suite.test_oven()
+		prepare("Debug")
+		--test.isequal(path.getabsolute("obj/Debug"), cfg.objdir)
+
+		prepare("Release")
+		--test.isequal(path.getabsolute("obj/Release"), cfg.objdir)
```
Comment (samsinsane) [inline tests/oven/test_compilebuildoutputstarget.lua]: I'm guessing these shouldn't be commented out?

Comment (tvandijck) [reply tests/oven/test_compilebuildoutputstarget.lua]: no, I just need to finish the test...
I copy pasted a test from somewhere else, and things were blowing up just in the oven itself already. so just the "prepare" calls was enough to set things off...

I didn't afterwards finish clean this up.

Comment (starkos) [reply tests/oven/test_compilebuildoutputstarget.lua]: The comment from @samsinsane is still valid: does this test need to be finished?

Comment (tvandijck) [reply tests/oven/test_compilebuildoutputstarget.lua]: yes and no... the test is valid without any real asserts or testing for values. The thing would crash during the oven part... so just forcing a bake was enough.

the commented out lines, I can remove if that's preferable..

**Comment by tvandijck** [comment]
what the code is trying to do, is automatically add files that would be generated by the buildcommand, to another project, with possibly mapped configurations.
```lua
compilebuildoutputs 'true'
compilebuildoutputstarget 'tag_edit'
```

**Comment by starkos** [comment]
See as @tvandijck has moved on, I'm going to go ahead and close this one. It needs a meaningful description of what it does, and unit tests to show that it does it. If someone wants to take this one on, [this comment](https://github.com/premake/premake-core/pull/968#issuecomment-387091797) might be helpful: *"automatically add files that would be generated by the buildcommand, to another project, with possibly mapped configurations"*. Sounds useful (but not at all what is suggested by the original title of "Refactor generated files")?

**Pull request closed without merging**